compiler/front/cli_reporter.nim

```diff
@@ -3597,63 +3574,19 @@ proc reportHook*(conf: ConfigRef, r: Report): TErrorHandling =
   ## Default implementation of the report hook. Dispatches into
   ## `reportBody` for report, which then calls respective (for each report
   ## category) `reportBody` overloads defined above
-  let tryhack = conf.m.errorOutputs == {}
-  # REFACTOR this check is an absolute hack, `errorOutputs` need to be
-  # removed. For more details see `lineinfos.MsgConfig.errorOutputs`
-  # comment
   assertKind r
 
-  if (r.kind == rdbgVmCodeListing) and (
-    (
-     # If special expand target is not defined, debug all generated code
-     ("expandVmListing" notin conf.symbols)
-    ) or (
-     # Otherwise check if listing target is not nil, and it's name is the
-     # name we are targeting.
-     (not r.debugReport.vmgenListing.sym.isNil()) and
-     (
-       r.debugReport.vmgenListing.sym.name.s ==
-       conf.symbols["expandVmListing"]
-  ))):
-    echo conf.reportFull(r)
+  let
+    wkind = conf.writabilityKind(r)
 
-  elif r.kind == rdbgVmCodeListing or (
-    # Optionally Ignore context stacktrace
-    not controller.semStack and r.kind == rdbgTraceLine
-  ):
+  if wkind == writeDisabled:
```

Comment:
Feels like if we got it right, then we should first and foremost case on the wkind, then whatever additional logic.

Reply:
debug reports can be both enabled and force enabled, and sem tracer first needs to be checked for the trace group rotation. So adding a case here is not really useful, since report writability kind does not necessarily dictate **how** it is written, just whether it **can/must/cannot** be written